File: src/LightNap.Core/Api/ApiResponseType.cs
namespace LightNap.Core.Api
{
    /// <summary>
    /// Represents the response types for an API.
    /// </summary>
    public enum ApiResponseType
    {
        /// <summary>
        /// The API call was successful.
        /// </summary>
        Success,

        /// <summary>
        /// An error occurred during the API call. User-friendly error messages are available in the <see cref="ApiResponseDto{T}.ErrorMessages"/> property.
        /// </summary>
        Error,

        /// <summary>
        /// An unexpected error occurred during the API call.
        /// </summary>
        UnexpectedError
    }
}

File: src/lightnap-ng/src/app/core/models/api/api-response-type.ts
/**
 * Represents the type of response received from an API call.
 */
export type ApiResponseType =
    /**
     * Indicates that the API call was successful.
     */
    "Success"
    /**
     * Indicates that the API call encountered an expected error with user-friendly error messages.
     */
    | "Error"
    /**
     * Indicates that the API call encountered an unexpected error.
     */
    | "UnexpectedError";
